src/common/legacy_config_opts.h

```diff
@@ -966,6 +964,7 @@ OPTION(bluestore_bluefs_reclaim_ratio, OPT_FLOAT) // how much to reclaim at a ti
 OPTION(bluestore_bluefs_balance_interval, OPT_FLOAT) // how often (sec) to balance free space between bluefs and bluestore
 // how often (sec) to dump allocation failure happened during bluefs rebalance
 OPTION(bluestore_bluefs_balance_failure_dump_interval, OPT_FLOAT)
+OPTION(bluefs_bdev_discard, OPT_STR)
```

Comment:
I'll consider your opinion after this patch become stable and modified to resolve other performance issue.
Thanks!

Reply:
I think the main issues of this patch are three.
1. extensive memory usage, due to keeping locations to discard.
2. too much commands to ssd.
3. assertion failure. (and cannot be compiled.)


@ifed01 
My Thoughts about the issues.
1. 
- I'll check the problem about memory usage. But, I guessed the maximum memory usage of this patch was twice as much as that of existing StupidAllocator. That's because I thought it kept duplicate information of existing allocator. I just thought discarding list was kept like free list in StupidAllocator. I'll check the specific cause of extensive memory. Fragmentation of intervals of discarding list might be problem...
- If the intervals of discarding list is allocated first, i think the interval set should be implemented hierarchically by their intervals' length for fast search, like existing method...

2. 
- Discarding command can be issued considering current number of in-flight requests.

3. 
- It should be resolved as soon as possible with detailed debugging

Thanks for your detailed opinion

Personal comments:
The period of pending this pr is too long.
I'll fix these issues and remove the [DNM] header in this PR, as soon as possible.